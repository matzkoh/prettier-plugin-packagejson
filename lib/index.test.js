const { readFile } = require('fs/promises')
const { beforeAll, describe, expect, it } = require('@jest/globals')
const prettier = require('prettier')
const { parsers } = require('prettier/parser-babel')
const { testPath } = require('.')

describe('testPath()', () => {
  it.each(['package.json', 'foo/bar/package.json'])('matches: %s', path => {
    expect(testPath(path)).toBe(true)
  })

  it.each([
    'Package.json',
    'package.JSON',
    'package-lock.json',
    'composer.json',
    'package.json/composer.json',
    'foo.json',
    'bar.js',
    undefined,
  ])('not matches: %s', path => {
    expect(testPath(path)).toBe(false)
  })
})

describe('format', () => {
  let uglyJson

  beforeAll(async () => {
    uglyJson = await prettier.format(
      await readFile(require.resolve('./fixtures/ugly.json'), 'utf8'),
      {
        filepath: 'package.json',
      },
    )
  })

  it('should be correct', () =>
    expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath: 'package.json',
          plugins: ['./lib/index.cjs'],
        }),
      ),
    ).resolves.toMatchSnapshot())

  it('should not format if not a package.json', () =>
    expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          parser: 'json-stringify',
          filepath: 'foo.json',
          plugins: ['./lib/index.cjs'],
        }),
      ),
    ).resolves.toBe(uglyJson))

  it('preprocess should be executed', async () => {
    const preprocess = jest.fn(() => '{}')
    parsers['json-stringify'].preprocess = preprocess

    await expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath: 'package.json',
          plugins: ['./lib/index.cjs'],
        }),
      ),
    ).resolves.toBe('{}\n')

    delete parsers['json-stringify'].preprocess
  })

  it('should follow custom order', () =>
    expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath: 'package.json',
          plugins: ['./lib/index.cjs'],
          packageSortOrder: ['version', 'name'],
        }),
      ),
    ).resolves.toMatchSnapshot())
})
