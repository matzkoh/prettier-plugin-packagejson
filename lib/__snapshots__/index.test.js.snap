// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format should be correct 1`] = `
"{
  "name": "",
  "version": "",
  "description": "",
  "keywords": [
    "C",
    "B",
    "A"
  ],
  "license": "",
  "author": "",
  "scripts": {
    "prebuild": "",
    "build": "",
    "postbuild": "",
    "dev": "",
    "lint": "",
    "start": "",
    "pretest": "",
    "test": "",
    "posttest": ""
  },
  "dependencies": {
    "A": "*",
    "B": "*"
  },
  "devDependencies": {
    "A": "*",
    "B": "*"
  },
  "peerDependencies": {
    "A": "*",
    "B": "*"
  },
  "optionalDependencies": {
    "A": "*",
    "B": "*"
  }
}
"
`;

exports[`format should follow custom order 1`] = `
"{
  "version": "",
  "name": "",
  "description": "",
  "keywords": [
    "C",
    "B",
    "A"
  ],
  "license": "",
  "author": "",
  "scripts": {
    "prebuild": "",
    "build": "",
    "postbuild": "",
    "dev": "",
    "lint": "",
    "start": "",
    "pretest": "",
    "test": "",
    "posttest": ""
  },
  "dependencies": {
    "A": "*",
    "B": "*"
  },
  "devDependencies": {
    "A": "*",
    "B": "*"
  },
  "peerDependencies": {
    "A": "*",
    "B": "*"
  },
  "optionalDependencies": {
    "A": "*",
    "B": "*"
  }
}
"
`;
